homeassistant/components/linky/config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Config flow to configure the Linky integration."""
+import logging
+
+import voluptuous as vol
+from pylinky.client import LinkyClient, PyLinkyError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_TIMEOUT, CONF_USERNAME
+from homeassistant.core import callback
+
+from .const import DEFAULT_TIMEOUT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.error("LINKY_CONFIG_FLOW")
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class LinkyFlowHandler(config_entries.ConfigFlow):
+    """Handle a Linky config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize Linky config flow."""
+        self._username = None
+        self._password = None
+        self._timeout = None
+
+    def _configuration_exists(self, username: str) -> bool:
+        """Return True if username exists in configuration."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if entry.title == username:
+                return True
+        return False
+
+    @callback
+    def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
+                    ): str,
+                    vol.Required(
+                        CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
+                    ): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+
+        if user_input is None:
+            return await self._show_setup_form(user_input, None)
+
+        self._username = user_input[CONF_USERNAME]
+        self._password = user_input[CONF_PASSWORD]
+        self._timeout = user_input.get(CONF_TIMEOUT, DEFAULT_TIMEOUT)
+
+        if self._configuration_exists(self._username):
+            errors = {}
+            errors[CONF_USERNAME] = "username_exists"
+            return await self._show_setup_form(user_input, errors)
+
+        client = LinkyClient(self._username, self._password, None, self._timeout)
+        try:
+            self.hass.async_add_executor_job(client.login)
+            client.fetch_data()
```

Comment:
Fetching the data also does I/O so needs to be done in the executor thread pool.